src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,124 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
 import './TodoList.scss';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+export class TodoList extends React.Component {
+  state = {
+    todo: '',
+    criterion: '',
+    selectedTodo: 0,
+  };
+
+  handleInput = (event) => {
+    this.setState({ todo: event.target.value });
+  };
+
+  handleSelect = (event) => {
+    this.setState({ criterion: event.target.value });
+  };
+
+  selectUser = (userId, todoId) => {
+    this.setState({ selectedTodo: todoId });
+    this.props.selectUser(userId);
+  };
+
+  render() {
+    const { todos, listHasLoaded } = this.props;
+    const { todo, criterion, selectedTodo } = this.state;
+
+    const preparedTodos = todos
+      .filter((item) => {
+        switch (criterion) {
+          case 'active':
+            return !item.completed;
+          case 'completed':
+            return item.completed;
+          default:
+            return true;
+        }
+      })
+      .filter((item) => {
```

Comment:
For the future - better to make all the filtering on the upper level and pass here only ALREADY prepared todos